examples/role-inline-policy-auto-name/index.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as aws from "@pulumi/aws";
+
+const role = new aws.iam.Role('testrole', {
+  assumeRolePolicy: aws.iam.getPolicyDocumentOutput({
+    statements: [
+      {
+        actions: ['sts:AssumeRole'],
+        principals: [
+          {
+            type: 'Service',
+            identifiers: ['lambda.amazonaws.com'],
+          },
+        ],
+      },
+    ],
+  }).json,
+  inlinePolicies: [
+    {
+      policy: aws.iam.getPolicyDocumentOutput({
+        statements: [
+          {
+            actions: ['s3:GetObject'],
+            resources: ['*'],
+          },
+        ],
+      }).json,
+    },
+  ],
+});
+
+const role2 = new aws.iam.Role('testrole-with-empty-inline', {
+  assumeRolePolicy: aws.iam.getPolicyDocumentOutput({
+    statements: [
+      {
+        actions: ['sts:AssumeRole'],
+        principals: [
+          {
+            type: 'Service',
+            identifiers: ['lambda.amazonaws.com'],
+          },
+        ],
+      },
+    ],
+  }).json,
+  inlinePolicies: [{}],
```

Comment:
This is a great example, I guess with this change this will be equivalent to `inlinePolicies: []`. It looks kind of clunky. To double check, the motivation for this is that we need to support programs that are out there that are already written like this, or some other motivation? For users writing new code do we want to direct them to writing `inlinePolicies: []` or even omitting  `inlinePolicies` entirely?

Reply:
Yep that's correct. I can't think of why as a developer you would want to provide an empty policy like that, but I've learned the hard way that there will probably be someone out there that does it.